### PR TITLE
Switched over to using aws-sdk default promise implementation

### PR DESCRIPTION
**Pull request by depowell-D2L**
This mainly involved removing `Async` from calls and calling `.promise()`
Had to do one work-around for `s3.getSignedUrl` doesn't have a `.promise()`
Removed unneeded `promisifyAll` calls and imports of 'bluebird'

I would recommend viewing the diff with white-space hidden (add `?w=1` to the url)

### CODE REVIEW

File: src/util/file-storage.js [hunk 1]
```diff
@@ -134,14 +137,26 @@ class FileStorage {
...
-					{
-						Bucket: this.bucket,
-						Key: key
-					});
+				const getSignedUrlParams = {
+					Bucket: this.bucket,
+					Key: key
+				};
+
+				// Workaround due to lack of promise(), see https://github.com/aws/aws-sdk-js/issues/1008
```
Comment (D2LChrisC) [inline src/util/file-storage.js]: Lame. :(

File: tests/util/file-storage.tests.js [hunk 2]
```diff
@@ -26,24 +22,24 @@ const storage = new FileStorage(
 	'http://localhost:4569',
 	encryptionKeyId);
 
-Bluebird.promisifyAll(s3);
-
 describe('File storage utility', () => {
 
 	before(done => {
-		s3.createBucketAsync({ Bucket: testBucket })
+		s3.createBucket({ Bucket: testBucket })
```
Comment (D2LChrisC) [inline tests/util/file-storage.tests.js]: In hindsight, the promises actually make this uglier. :(
```suggestion
		s3.createBucket({ Bucket: testBucket }, done);
```

File: tests/util/file-storage.tests.js [hunk 3]
```diff
@@ -26,24 +22,24 @@ const storage = new FileStorage(
...
+			.promise()
 			.then(() => done())
 			.catch(err => done(err));
 	});
 
 	after(done => {
-		s3.deleteBucketAsync({ Bucket: testBucket })
-			.catch(() => { })
-			.finally(() => done());
+		s3.deleteBucket({ Bucket: testBucket })
```
Comment (D2LChrisC) [inline tests/util/file-storage.tests.js]: Same here.
```suggestion
		s3.deleteBucket({ Bucket: testBucket }, done);
```

File: src/util/file-storage.js (hunk 1 above)
Comment (depowell-D2L) [reply src/util/file-storage.js]: I know. It's one of the few that doesn't have it yet

File: tests/util/file-storage.tests.js (hunk 3 above)
Comment (D2LChrisC) [reply tests/util/file-storage.tests.js]: Huh... weird. No, first I heard about it. Does the call work? It just fails after the bucket has been deleted?

File: tests/util/file-storage.tests.js [hunk 4]
```diff
@@ -26,24 +22,37 @@ const storage = new FileStorage(
...
-			.finally(() => done());
+		s3.listObjects({ Bucket: testBucket })
+			.promise()
+			.then((objects) => {
+				const contents = objects.Contents;
+				let promises = [];
+
+				for (let i = 0; i < contents.length; i += 1) {
+					promises += s3
+						.deleteObject({
```
Comment (D2LChrisC) [inline tests/util/file-storage.tests.js]: Oh man, not even a bulk delete call? Amazon, srsly!?

Comment (depowell-D2L) [reply tests/util/file-storage.tests.js]: Yup :(

Comment (depowell-D2L) [reply tests/util/file-storage.tests.js]: Wait NM. I just can't read :(. There is a deleteObjects. I'll adjust that

**Pull request merged**